app/components/homepage-components/Testimonials.tsx

```diff
@@ -0,0 +1,129 @@
+"use client";
+
+import React, { useEffect, useRef, useState } from "react";
+import Image from "next/image";
+import TestimonialCard from "./TestimonialCard";
+import { useTheme } from "next-themes";
+
+import { testimonials } from "@/constants";
+import {
+  arrowLeft,
+  arrowRight,
+  arrowLeftDark,
+  arrowRightDark,
+} from "@/public/svg-icons";
+
+interface ButtonsProps {
+  additionalClassesLeft: string;
+  additionalClassesRight: string;
+}
+
+const Testimonials = () => {
+  const { theme } = useTheme();
+  const [current, setCurrent] = useState(0);
+
+  const intervalIdRef = useRef<number | null>(null);
+
+  const setAutoAdvance = () => {
+    if (intervalIdRef.current) clearInterval(intervalIdRef.current);
+    intervalIdRef.current = window.setInterval(() => {
+      next();
+    }, 5000);
+  };
+
+  useEffect(() => {
+    setAutoAdvance();
+    return () => {
+      if (intervalIdRef.current) clearInterval(intervalIdRef.current);
+    };
+  }, []);
+
+  useEffect(() => {
+    setAutoAdvance();
+    return () => {
+      if (intervalIdRef.current) clearInterval(intervalIdRef.current);
+    };
+  }, [current]);
```

Comment:
Looking great! I am only making this suggestion to provide you with the most value. What you have works fine... Another approach you could consider here is to extract the auto-advancing logic into a custom hook. 

I will reference a simple guide below that should get you started if you decide to implement it.

```
import { useEffect, useRef } from 'react';

function useAutoAdvance(callback, delay) {
  const intervalIdRef = useRef(null);

  useEffect(() => {
    intervalIdRef.current = setInterval(callback, delay);
    return () => clearInterval(intervalIdRef.current);
  }, [callback, delay]);

  return intervalIdRef;
}
```

// In your component:
`const intervalIdRef = useAutoAdvance(next, 5000);`